src/apitest/buffer_set_lines.c

```diff
@@ -150,6 +150,35 @@ MU_TEST(test_modified_is_set)
   char_u *lines[] = {"one"};
   vimBufferSetLines(curbuf, 0, 0, lines, 1);
   mu_check(vimBufferGetModified(curbuf) == TRUE);
+}
+ MU_TEST(set_lines_into_empty_buffer)
+{
+  buf_T *buf = vimBufferOpen(NULL, 0, 0);
+
+  char_u* lines[] = {
+    "line 1",
+    "line 2",
+    "line 3",
+    "line 4",
+    "line 5",
+    "line 6"
+  };
+
+  int i;
+
+  vimBufferSetLines(buf, 1, 6, lines, 6);
+
+  for (i = 0; i < 6; ++i) {
+    printf("LINE %i: %s\n", i, vimBufferGetLine(buf, i));
+  }
+
+  mu_check(strcmp(vimBufferGetLine(buf, 1), "line1") == 0);
+  mu_check(strcmp(vimBufferGetLine(buf, 2), "line2") == 0);
+  mu_check(strcmp(vimBufferGetLine(buf, 3), "line3") == 0);
+  mu_check(strcmp(vimBufferGetLine(buf, 4), "line4") == 0);
+  mu_check(strcmp(vimBufferGetLine(buf, 5), "line5") == 0);
+  mu_check(strcmp(vimBufferGetLine(buf, 6), "line6") == 0);
```

Comment:
Oh, and this should be "line 1" not "line1".

Reply:
ah yes, of course. thanks.